resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -172,9 +175,15 @@ private[spark] class KubernetesClusterSchedulerBackend(
         runningExecutorsToPods.remove(executorId).map { pod =>
           kubernetesClient.pods().delete(pod)
           runningPodsToExecutors.remove(pod.getMetadata.getName)
+          failedExecutors -= pod.getMetadata.getName
         }.getOrElse(logWarning(s"Unable to remove pod for unknown executor $executorId"))
       }
     }
+
+    // e.g. if we expect to create 2 executor but every executor failed,
```

Comment:
Javadoc needs a bit of rewording to explain what the return value represents and how it is calculated.

Reply:
how about this 
```
    // It represent current created executors exclude failed one.
    // To avoid to create too many failed executor,
    // we limit the accounting size of failed executors to totalExpectedExecutors
    // so after create 2*totalExpectedExecutors executors,
    // we stop create more even if all of them failed
```